pkg/pod/pod.go

```diff
@@ -394,6 +408,15 @@ func isPodReadyImmediately(featureFlags config.FeatureFlags, sidecars []v1beta1.
 	return true
 }
 
+func isDefaultForbiddenEnv(envName string, defaultForbiddenEnvs []string) bool {
```

Comment:
NIT:

Go 1.18 introduced this experimental package to check if an element exists in an array of strings - https://golangtutorial.dev/tips/golang-slice-contains-method/

We could try utilizing it in future.